packages/flutter_tools/test/integration/flutter_run_integration_test.dart

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:test/test.dart';
+import 'package:vm_service_client/vm_service_client.dart';
+
+import '../src/context.dart';
+import 'flutter_test_driver.dart';
+import 'util.dart';
+
+Directory _tempDir;
+FlutterTestDriver _flutter;
+
+void main() {
+
+  setUp(() async {
+    _tempDir = await fs.systemTempDirectory.createTemp('test_app');
+    await _setupSampleProject();
+    _flutter = new FlutterTestDriver(_tempDir); 
+  });
+
+  tearDown(() async {
+    try {
+      await _flutter.stop();
+      _tempDir?.deleteSync(recursive: true);
+      _tempDir = null;
+    } catch (e) {
+      // Don't fail tests if we failed to clean up temp folder.
+    }
+  });
+
+  Future<VMIsolate> breakInBuildMethod(FlutterTestDriver flutter) async {
+    return _flutter.breakAt(
+      fs.path.join(_tempDir.path, 'lib', 'main.dart'),
+      9
+    );
+  }
+
+  Future<VMIsolate> breakInTopLevelFunction(FlutterTestDriver flutter) async {
+    return _flutter.breakAt(
+      fs.path.join(_tempDir.path, 'lib', 'main.dart'),
+      17
+    );
+  }
+
+  group('FlutterTesterDevice', () {
+
+    testUsingContext('can hot reload', () async {
```

Comment:
If all the integration tests need the same setup and teardown, perhaps have a `testIntegration` test entrypoint? (instead of `test()` or `testUsingContext()`).

Also, are we using any of the functionality from `testUsingContext`? From memory, that setup may be more useful to the flutter_tools unit tests.

Reply:
Or, have integration tests defining in a class, with setup, test, and teardown methods subclasses can override? That could also be a place to put common utility functions, like `evaluateTrivialExpressions`